Src/ILGPU/IR/Types/TypeNode.cs

```diff
@@ -301,5 +301,27 @@ public T As<T>(ILocation location)
         public override string ToString() => ToPrefixString();
 
         #endregion
+
+        #region Operators
+
+        /// <summary>
+        /// Returns true if the left and right type nodes are the same.
+        /// </summary>
+        /// <param name="left">The left type node.</param>
+        /// <param name="right">The right type node.</param>
+        /// <returns>True, if the left and right type nodes are the same.</returns>
+        public static bool operator ==(TypeNode left, TypeNode right) =>
```

Comment:
I am not sure whether we should really implement these operators as the compiler is based on the invariant that if the types are the "same" (using `==`) they actually represent the same type. Is this really required?

Reply:
I've changed the PR to handle this differently. It only affect the one method in `StructureValues.cs`, so I've just changed the assertion.